src/middlewared/middlewared/api/base/decorator.py

```diff
@@ -68,6 +69,10 @@ def api_method(
     This is incompatible with `roles`. Additional review will be required in order to validate that its use complies
     with security standards.
 
+    `inject_lzh` if set to True, will inject the thread-local storage object with an open libzfs lightweight handle
+    as the first argument to the decorated method. NOTE: this is using a traditional threading.local() object and
+    not a ContextVar so the method must be a non-coroutine based method/function.
```

Comment:
Can we use `if asyncio.iscoroutinefunction(func):` check to prevent using `inject_lzh` to coroutines?

Reply:
Yep, I'll do that.